core/src/main/java/com/datastax/oss/driver/internal/core/metadata/DefaultEndPoint.java

```diff
@@ -65,20 +62,8 @@ public String asMetricPrefix() {
     return metricPrefix;
   }
 
-  private static String buildMetricPrefix(InetSocketAddress addressAndPort) {
-    StringBuilder prefix = new StringBuilder();
-    InetAddress address = addressAndPort.getAddress();
-    int port = addressAndPort.getPort();
-    if (address instanceof Inet4Address) {
-      // Metrics use '.' as a delimiter, replace so that the IP is a single path component
-      // (127.0.0.1 => 127_0_0_1)
-      prefix.append(address.getHostAddress().replace('.', '_'));
-    } else {
-      assert address instanceof Inet6Address;
-      // IPv6 only uses '%' and ':' as separators, so no replacement needed
-      prefix.append(address.getHostAddress());
-    }
+  private static String buildMetricPrefix(InetSocketAddress address) {
     // Append the port since Cassandra 4 supports nodes with different ports
-    return prefix.append(':').append(port).toString();
+    return address.getHostString().replace('.', '_') + ':' + address.getPort();
```

Comment:
Note that, unlike the previous implementation, this will use the host name if that's what was provided (even for resolved addresses).
If users provide their contact points by name, I think it makes sense to use that for metrics too.

Reply:
I see that `address.getHostString()` can return null, at least in theory (although it seems to be a pathological case). Do we want to defensively prevent against NPEs here?

Reply:
Maybe you can reuse class that I've created to format addresses in Insights related code:
```

class AddressFormatter {

  static String nullSafeToString(Object address) {
    if (address instanceof InetAddress) {
      return nullSafeToString((InetAddress) address);
    } else if (address instanceof InetSocketAddress) {
      return nullSafeToString((InetSocketAddress) address);
    } else if (address instanceof String) {
      return address.toString();
    } else {
      return "";
    }
  }

  static String nullSafeToString(InetAddress inetAddress) {
    return inetAddress != null ? inetAddress.getHostAddress() : null;
  }

  static String nullSafeToString(InetSocketAddress inetSocketAddress) {
    if (inetSocketAddress != null) {
      if (inetSocketAddress.isUnresolved()) {
        return String.format(
            "%s:%s",
            nullSafeToString(inetSocketAddress.getHostName()), inetSocketAddress.getPort());
      } else {
        return String.format(
            "%s:%s", nullSafeToString(inetSocketAddress.getAddress()), inetSocketAddress.getPort());
      }
    }
    return null;
  }
}
```

Reply:
That class does not exist in the OSS driver. Also one additional requirement is to replace dots in the generated string.

Reply:
@adutra yeah, I don't see how it can happen. The method is called directly from the constructor, so I'll just throw an IAE just in case.

Reply:
> That class does not exist in the OSS driver. Also one additional requirement is to replace dots in the generated string.

I mean that we could cherry-pick that class.

Reply:
Should be good for now, it's just 2 lines of code and not exactly the same (replacing the dots + use getHostString instead of getHostName).